### DISCUSSION TITLE
Issues with internet interface

### DISCUSSION THREAD

**Discussion by doobes** [TriliumNext/Trilium#6054] [Ideas]
So,

I've upgraded my server and all my devices to the latest version.

The installed app's work fine.

The browser based ones, not so much, both on computers and android devices.

This is a screenshot of the app and the browser interface:

![image](https://github.com/user-attachments/assets/fa4ee05d-6a9b-4d20-a2d5-8792bb3054c2)

As you can see there is a discrepancy here. 

When I refresh the browser version I get a blob complaint about notes not found. I also cannot close the tabs seen in the browser version.

I've managed to corrupt the database attempting to delete notes on my portable devices to the point that I had to ssh into the server and reboot it.

Let me know if I can provide more information.

Cheers

**Comment by doobes** [comment]
More:

![image](https://github.com/user-attachments/assets/7811e42c-ad1d-46e1-8621-babcb3877465)

**Comment by perfectra1n** [comment]
Just trying to understand:

1. On your server you went from the latest Zadam container to some TriliumNext container 
2. You deleted some notes on your mobile device using the web interface, and had to reboot the server because it stopped responding?
3. Now when navigating to the web interface of the upgraded server, after the reboot, it says that `Blob .... was not found`?

Some questions:

1. What container version / SHA of TriliumNext are you running?
2. Do you have any logs of the container before you rebooted the entire server? If you're using your CIFS / SMB for this container, to me this sounds like your SQLite lost connection or couldn't keep up with your changes. Please add the `nobrl` mount option which is required for SQLite. I'm wondering if you ran into an issue as described [here](https://lchenghui.com/nobrl-for-mount-cifs). Running databases over network filesystems such as SMB / NFS is a hornet's nest of issues. I would strongly suggest running the DB locally (having the Trilium database local to the machine it's running on), or something else like iSCSI.
3. Can you share the logs of the server, from boot to when the error happens (when you navigate to the note)?
4. I wonder if any of the options in the `Advanced` tab might help you:
![image](https://github.com/user-attachments/assets/8ced572a-77ae-4599-b965-0e709e540993)

**Comment by doobes** [comment]
Greetings!

My server is set up per our discussions here: [(https://github.com/orgs/TriliumNext/discussions/396)]

I had not seen your message yesterday so, I re-initialized everything.

I exported all my files from the Windows client and then shut it down.  I brought the docker container down on the server. I manually deleted the files from the share, and the local storage on all the machines I'm using Trilium Next on.

I brought the docker container up and let it initialize.

I started up each of the individual clients and re-connected to the server.

I then imported the stuff I exported without doing anything more.

This is the difference between the Web client and the Application:

![image](https://github.com/user-attachments/assets/8049e476-1b16-48ce-825c-49d66f6d2202)

The web client has two notes the Application doesn't.

I deleted one of these notes in the browser and refreshed the page. The first thing that happened is I got a bunch of error flags, and then it complained it had lost connection. It then popped up the login page, where I had to provide my password again.

In the Application, the sync icon is spinning without end. It's been spinning the entire time it has taken to craft this note. Last time this occurred, I had to reboot the server.

So, something is obviously not right.

I'll leave everything in it's current place to see if I can provide you any data to help.

My gut feel at this instance is this is indeed an artifact of the network share. When I was writing locally to the pi, I had none of these issues. If this is the case, then this would definitely be something to be resolved longterm as it would be a major showstopper for anything more than a casual deployment. I could plug a thumbdrive into the Rpi, but honestly, that's a pretty backwoods solution.

Appreciate the feedback.

**Reply by perfectra1n** [reply]
The Raspberry Pi would be a solid solution, assuming your SD card is large enough (Trilium doesn't need much space at all). You could then use Kopia or something of the sort to then copy the backups to your SMB share for increased resiliency.

**Reply by doobes** [reply]
Define not much space.

I've really not started doing anything of significance in Trilium.

I'm using Trilium as a replacement for OneNote. 

I used OneNote to build project logs with collections of all the information I collect in my developments.  I'm a maker, so this is data sheets for vendor info, drawing scraped from the web, etc, etc, etc. My OneNote folders got pretty big.

Thus the desire to store everything on my Synology box. The prime reason for moving away from OneNote was that I was not in possession of my data. The multi-OS capability of Trilium was a nice to have. 

Lemme ask a semi-related question? Would it make sense to host Trilium on the Synology? 

I've not done anything on the Synology other than the "native" apps.  Definitely no Docker stuff. I'm a bit hesitant to jump into this as if I buggered something up, there is a real risk of data loss. While I've got the Synology backed up, I wouldn't want to test that from my own stupidity.

Another real issue with this is opening up my firewall to the Synology box.  It's one thing to port forward to a Rpi, a whole different topic to port forward to the NAS. Lot's more at risk.  I know enough about this stuff to be dangerous...

In playing with Docker on the Rpi, I had to go through the process numerous times to get it right, which included a couple of reformat the SD card episodes.

Appreciate the feedback.

**Reply by perfectra1n** [reply]
Well you're in luck because I'm a pentester by profession. I can assure you that if you port forward to *just* Trilium running on your Synology NAS, you're fine. Hackers and the like also don't mess much with the common individual, because there just isn't that much to gain from an individual. They go after larger businesses. I've also audited Trilium pretty heavily to make sure there's nothing I can even find in the application or code. Of course data loss is a real threat, but I'd hope that Synology's GUI / CLI tools could help you take regular backups, and if not, a cronjob running Kopia to another location (even on your disk) would be more than enough.

From what I've gathered by others such as @maphew, they run Trilium on their Synology OS using Docker. Perhaps they can offer some insights, but I would definitely suggest running it on your Synology directly :)

My Trilium has quite a few PDFs, images, etc. (binary objects taking up lots of space) and my DB is only ~500MB. That's not a ton of space, especially if I can assume that you're using a very common 32GB SD card for the Pi.

**Comment by doobes** [comment]
Just a note.  The desktop client just finished synchronizing.

**Comment by perfectra1n** [comment]
Are you able to share any logs from your Trilium container? From container start to error?

**Reply by doobes** [reply]
Well,

The docker had been stood up with the -d switch.

So, I just shut it down and then started without the -d switch.

It's fairly slow.  Far slower than I've ever seen.

Firing up the desktop app, it took about 60 seconds to sync without any changes from last time.

I'm using a different browser this time and everything display's as synchronized in the browser version.

Could this be a browser cache issue?

Anyway, here a grab of the ssh output

https://pastebin.com/5HW357TA

Looking through, there are a couple of errors.

cheers

**Comment by doobes** [comment]
Well,

This seems to be isolated to interaction on my portable devices.

I just had it log me out of my phone:

![image](https://github.com/user-attachments/assets/4b2f3584-7abe-40f1-8368-d867e97f413e)

For laughs, I changes to an NFS mount.  Got it working this time ;)

I'll report back
